panther_lights/src/controller_node.py

```diff
@@ -323,99 +324,124 @@ def _get_image_animation_description(self, animation: LEDImageAnimation) -> dict
 
-                    try:
-                        BASIC_ANIMATIONS[anim_desc['type']]
-                    except KeyError as err:
-                        raise KeyError(f'Undefined animation type: {err}')
+                if not 'type' in anim_desc:
+                    raise KeyError('Missing \'type\' in animation description')
 
-                    if panel == 'both':
-                        animation.front = BASIC_ANIMATIONS[anim_desc['type']](
-                            anim_desc, self._num_led, self._controller_frequency
-                        )
-                        animation.rear = BASIC_ANIMATIONS[anim_desc['type']](
-                            anim_desc, self._num_led, self._controller_frequency
-                        )
-                        break
-                    elif panel == 'front':
-                        animation.front = BASIC_ANIMATIONS[anim_desc['type']](
-                            anim_desc, self._num_led, self._controller_frequency
-                        )
-                    elif panel == 'rear':
-                        animation.rear = BASIC_ANIMATIONS[anim_desc['type']](
-                            anim_desc, self._num_led, self._controller_frequency
-                        )
-                    else:
-                        raise KeyError(f'Invalid panel type: {panel}')
+                try:
+                    BASIC_ANIMATIONS[anim_desc['type']]
+                except KeyError as err:
+                    raise KeyError(f'Undefined animation type: {err}')
 
-                if not hasattr(animation, 'front') or not hasattr(animation, 'rear'):
-                    raise KeyError(
-                        'Missing \'both\' or \'front\'/\'rear\' in animation description'
+                if panel == 'both':
+                    animation.front = BASIC_ANIMATIONS[anim_desc['type']](
+                        anim_desc, self._num_led, self._controller_frequency
                     )
+                    animation.rear = BASIC_ANIMATIONS[anim_desc['type']](
+                        anim_desc, self._num_led, self._controller_frequency
+                    )
+                    break
+                elif panel == 'front':
+                    animation.front = BASIC_ANIMATIONS[anim_desc['type']](
+                        anim_desc, self._num_led, self._controller_frequency
+                    )
+                elif panel == 'rear':
+                    animation.rear = BASIC_ANIMATIONS[anim_desc['type']](
+                        anim_desc, self._num_led, self._controller_frequency
+                    )
+                else:
+                    raise KeyError(f'Invalid panel type: {panel}')
 
-                if 'name' in anim:
-                    animation.name = anim['name']
+            if not hasattr(animation, 'front') or not hasattr(animation, 'rear'):
+                raise KeyError('Missing \'both\' or \'front\'/\'rear\' in animation description')
```

Comment:
This will never be reached ad all it does is checking if object of class animation has predefined keys which it obviously does

Reply:
That is not true, if attribute is defined but not assigned it `hasattr` will return false.